File: clojure/16.clj
;; 2^15 = 32768 and the sum of its digits is 3 + 2 + 7 + 6 + 8 = 26.
;;
;; What is the sum of the digits of the number 2^1000?

(load "util")

(defn euler-16 [n m]
  (sum (digits-of (exp n m))))

(run (euler-16 2 1000))

File: haskell/16.hs
-- 215 = 32768 and the sum of its digits is 3 + 2 + 7 + 6 + 8 = 26.
--
-- What is the sum of the digits of the number 2^1000?

import Util

euler16 = sum . toDigits

main = run $ euler16 (2^1000)
